src/SymfonyCache/PurgeListener.php

```diff
@@ -81,7 +93,20 @@ public function handlePurge(CacheEvent $event)
         }
 
         $response = new Response();
-        if ($event->getKernel()->getStore()->purge($request->getUri())) {
+        $store = $event->getKernel()->getStore();
+
+        if ($request->headers->has($this->purgeTagsHeader)
```

Comment:
why do we do tag invalidation in the same listener as purging? i feel like it should be a separate listener. in varnish, its glued into the ban handling because its a ban operation. for httpcache it seems to make sense to consider it not a BAN because its not a blacklist but a direct action onto the cache store. but purge sounds wrong to me too, because the path of the tag invalidation request does not matter. could we use a separate HTTP method like PURGETAGS? (this is an open question, i am not convinced it would be the best. but the mixup looks odd to me here)

Reply:
Well, I can just tell what my idea behind this was: `PURGE` is not an official HTTP method by specification. If I was to pick one that exists, I would've probably gone with `DELETE`. Anyway, the spec allows to invent custom ones so yeah, we could use `PURGETAGS`. But then again, to purge, means to clean and so I checked the docs of Varnish (which I personally consider to be somewhat a market leader here) and it says:

> A purge is what happens when you pick out an object from the cache and discard it along with its variants. Usually a purge is invoked through HTTP with the method PURGE.

And "tagging" to me is an object from the cache so imho it's perfectly fine. Also having it in the same listener is simple, doesn't change any behavior at all if you don't send that header, keeps things in place. I really like it where it is to be honest.

Reply:
my problem is that just depending on the tags header, this either uses the url to purge a specific page, or it uses the header to purge all tagged things, and ignores the url. so basically the two things have no code in common. it shows in the nested if/elseif/else. i would prefer a separate listener for that reason.

for the naming i agree that the term purge is correct. but as the semantics of the purge payload is completely different, i'd rather use a separate http method name.

Reply:
Okay, it's now in it's own `PurgeTagsListener` and the docs are updated accordingly.